plugins/async-node/core/src/index.ts

```diff
@@ -151,8 +153,12 @@ export class AsyncNodePluginPlugin implements AsyncNodeViewPlugin {
         let resolvedNode;
         if (this.isAsync(node)) {
           const mappedValue = this.resolvedMapping.get(node.id);
+          if (this.pendingUpdates.has(node.id)) {
+            return; // Skip processing if already scheduled
+          }
           if (mappedValue) {
             resolvedNode = mappedValue;
+            this.pendingUpdates.add(node.id);
```

Comment:
**Intent of pendingUpdates**

Prevent Duplicate Processing: When an asynchronous node is being processed (e.g., fetching data or resolving content), it's marked in pendingUpdates. This prevents the same node from being processed multiple times if it's encountered again before the current processing completes. This is particularly useful in scenarios where nodes might be rapidly updated or re-rendered.
Optimize Performance: By avoiding duplicate processing of nodes, the application can save on unnecessary computations or network requests, leading to better performance.
Synchronization: It helps in synchronizing the state of the node between the time it is first processed and when it is updated. This ensures that updates to nodes are handled in an orderly manner, respecting the sequence of updates.

**Separation from resolvedMapping**

Different Purposes: resolvedMapping is used to store the final resolved state of an asynchronous node. In contrast, pendingUpdates is used to track nodes that are currently being processed or have updates pending. Once the processing of a node is complete, it's moved from pendingUpdates to resolvedMapping.
Lifecycle Management: pendingUpdates is more about managing the lifecycle and processing state of nodes, while resolvedMapping is about storing the resolved data. This separation allows for more granular control over node processing and rendering.
Concurrency Control: The separation allows the system to handle concurrent updates to nodes more effectively. While resolvedMapping holds the current resolved state, pendingUpdates ensures that concurrent processes do not interfere with each other.

In summary, pendingUpdates is a mechanism to track and manage the processing state of asynchronous nodes to prevent duplicate processing and optimize performance, while resolvedMapping stores the resolved state of these nodes for rendering. This separation is crucial for efficient and accurate state management of asynchronous operations within the application.

Reply:
I get where you're coming from, if we want to avoid all unnecessary processing, why is this added to the set only after there is already a mapped value? The existence of a mapped value implies the node has already been processed and the final result is ready to be used, so at this point, [this](https://github.com/player-ui/player/pull/404/files#diff-9b846b61d9cdd8ebb4e14c47525e1fbc6d69e6088fd7d938d62577a72feb3327R156) is no longer a useful check.

I think you want to track this, you'd want to track it right away, instead of at the end